### PR TITLE
Add validate_policies function for schema validation

### PR DESCRIPTION
**Pull request by Iamrodos**
This PR adds a `validate_policies(policies, schema)` function that validates Cedar policies against a schema, exposing the cedar-policy `Validator` API to Python users.

## Motivation

Currently, validating Cedar policies against a schema requires shelling out to the Cedar CLI (`cedar validate`). This is inconvenient and adds an external dependency. By exposing the Validator API directly, users can:

- Validate policies programmatically in Python
- Integrate validation into CI/CD pipelines before deployment
- Catch policy errors (typos in entity types, invalid actions, type mismatches, unsafe access to optional attributes) before they cause silent failures at runtime

See the [Cedar validation documentation](https://docs.cedarpolicy.com/policies/validation.html) for details on what the validator checks.

## Usage

```python
from cedarpy import validate_policies, ValidationResult

policies = '''
    permit(
        principal == User::"alice",
        action == Action::"view",
        resource
    );
'''

schema = '''{ "": { "entityTypes": { "User": {}, ... }, "actions": { ... } } }'''

result = validate_policies(policies, schema)

if result.validation_passed:
    print("Policies are valid!")
else:
    for error in result.errors:
        print(f"Error: {error}")
```

## Design Decisions

The `validate_policies` function returns a `ValidationResult` for all error types (including parse errors) rather than raising `ValueError`. This differs from functions like `format_policies` which raise on invalid input.

Rationale: The purpose of a validator is to check validity. A validator that throws exceptions for invalid input wouldn't be useful as a validator. Users expect to call `validate_policies()` and check the result, not wrap it in try/except.

## Testing Approach

Three layers of testing ensure correctness:

1. **New unit tests** (`tests/unit/test_validate.py`) - 20 tests covering the API, error handling, schema formats, and edge cases.

2. **Rust CLI parity tests** - 8 of the unit tests use the same policy and schema files as the Rust CLI tests in `third_party/cedar/cedar-policy-cli/sample-data/`. This ensures our validation produces the same results as `cedar validate`.

3. **Integration test validation** - The existing integration tests now call `validate_policies` when the test definition specifies `shouldValidate: true`. This adds validation coverage to 22 additional test scenarios from the official Cedar integration test suite.

---

Implementation assisted by AI tooling; design decisions, testing strategy, and code review performed by the author.

### CODE REVIEW

**Comment by skuenzli** [comment]
Thank you for the PR @Iamrodos! I am at re:Invent now. I skimmed the code and the approach looks good. I will review it closely soon.

**Comment by Iamrodos** [comment]
@skuenzli Must say I am glad to not go to re:Invent anymore. I think I went for 9 or 10 years in a row. I think my first year was 2012. Have fun!

File: tests/integration/test_cedar_integration_tests.py [hunk 1]
```diff
@@ -55,8 +55,15 @@ def exec_authz_query_with_assertions(self,
...
-                                         should_validate: bool,  # ignored; currently don't have the equivalent
+                                         should_validate: bool,
                                          request_model: dict) -> None:
+        # Validate policies against schema if should_validate is True
+        if should_validate:
+            validation_result = cedarpy.validate_policies(policies, schema)
+            self.assertTrue(validation_result.validation_passed,
+                           f"Expected policies to validate but got errors: "
+                           f"{[str(e) for e in validation_result.errors]}")
+
```
Comment (skuenzli) [inline tests/integration/test_cedar_integration_tests.py]: 👍

File: src/lib.rs [hunk 2]
```diff
@@ -385,6 +403,90 @@ fn load_entities(entities_str: String, schema: Option<&Schema>) -> Result<Entiti
...
+                    policy_id: e.policy_id().to_string(),
+                    error: e.to_string(),
+                }
+            })
+            .collect(),
+    };
+
+    serde_json::to_string(&result).unwrap()
+}
+
```
Comment (skuenzli) [inline src/lib.rs]: 👍

File: tests/unit/test_validate.py [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+        policies = load_file_as_str(f"{self.SAMPLE_DATA}/sandbox_b/policies_6.txt")
+        schema = load_file_as_str(f"{self.SAMPLE_DATA}/sandbox_b/schema.json")
+
+        result = validate_policies(policies, schema)
+        self.assertTrue(result.validation_passed,
+                       f"Expected valid policy to pass. Errors: {[str(e) for e in result.errors]}")
+
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (skuenzli) [inline tests/unit/test_validate.py]: 🙌
